tracing/plugin/otlp.go

```diff
@@ -45,26 +45,27 @@ func init() {
 		InitFn: func(ic *plugin.InitContext) (interface{}, error) {
 			cfg := ic.Config.(*OTLPConfig)
 			if cfg.Endpoint == "" {
-				return nil, fmt.Errorf("otlp endpoint not set: %w", plugin.ErrSkipPlugin)
+				return nil, fmt.Errorf("OTLP endpoint not set: %w", plugin.ErrSkipPlugin)
 			}
 
-			opts := []otlptracegrpc.Option{
-				otlptracegrpc.WithEndpoint(cfg.Endpoint),
-				otlptracegrpc.WithDialOption(
-					grpc.WithBlock(),
-					grpc.WithReturnConnectionError(),
-				),
+			u, err := url.Parse(cfg.Endpoint)
+			if err != nil {
+				return nil, fmt.Errorf("OTLP endpoint %q is invalid: %w", cfg.Endpoint, err)
+			}
```

Comment:
It may make sense to check for [environment-based configuration](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md) here or attempt to create an exporter without providing the `WithEndpoint()` option to allow fallback to configuration from the environment.

Reply:
Seems we could use the environment variables by just removing our own check.

https://github.com/open-telemetry/opentelemetry-go/blob/931266483db80dc061b033c74f0db36a711ab117/exporters/otlp/otlpmetric/internal/otlpconfig/envconfig.go

@cpuguy83 @estesp What do you think? We are currently using the existence of the endpoint configuration as the way to enable tracing. Not checking the endpoint here removes the way to disable tracing.